File: token supply.py
class AefCoinToken {
    constructor() {
        this.totalSupply = 1000000; // Total supply of AEF Coin
        this.tokenSymbol = "AEF"; // Token symbol
        this.tokenName = "AEF Coin"; // Token name
        this.tokenHolders = {}; // Object to store token balances by wallet address
    }

    // Function to initialize a wallet with AEF Coins
    initializeWallet(walletAddress) {
        if (!this.tokenHolders[walletAddress]) {
            this.tokenHolders[walletAddress] = 0;
        }
    }

    // Function to transfer AEF Coins from one wallet to another
    transfer(senderAddress, receiverAddress, amount) {
        if (this.tokenHolders[senderAddress] >= amount) {
            this.tokenHolders[senderAddress] -= amount;
            this.tokenHolders[receiverAddress] += amount;
            return true; // Transfer successful
        }
        return false; // Insufficient balance
    }

    // Function to check the balance of a wallet
    getBalance(walletAddress) {
        return this.tokenHolders[walletAddress] || 0;
    }
}

// Usage example:

const token = new AefCoinToken();

// Initialize wallets
token.initializeWallet("Wallet1");
token.initializeWallet("Wallet2");

// Perform a transfer
if (token.transfer("Wallet1", "Wallet2", 100)) {
    console.log("Transfer successful!");
} else {
    console.log("Transfer failed: Insufficient balance.");
}

// Check wallet balances
console.log("Wallet1 Balance:", token.getBalance("Wallet1"));
console.log("Wallet2 Balance:", token.getBalance("Wallet2"));

File: token supply.js
class Token {
    constructor(name, symbol) {
        this.name = name;
        this.symbol = symbol;
        this.totalSupply = 0;
        this.holders = {};
    }

    // Mint new tokens and assign them to an address
    mint(receiver, amount) {
        if (!this.holders[receiver]) {
            this.holders[receiver] = 0;
        }
        this.totalSupply += amount;
        this.holders[receiver] += amount;
    }

    // Transfer tokens from one address to another
    transfer(sender, receiver, amount) {
        if (!this.holders[sender] || this.holders[sender] < amount) {
            return false; // Insufficient balance
        }
        if (!this.holders[receiver]) {
            this.holders[receiver] = 0;
        }
        this.holders[sender] -= amount;
        this.holders[receiver] += amount;
        return true;
    }

    // Check the balance of an address
    getBalance(address) {
        return this.holders[address] || 0;
    }

    // Get the total supply of tokens
    getTotalSupply() {
        return this.totalSupply;
    }
}

// Usage example:

const myToken = new Token("My Token", "MTK");

// Mint tokens for addresses
myToken.mint("Address1", 1000);
myToken.mint("Address2", 500);

// Transfer tokens between addresses
myToken.transfer("Address1", "Address2", 200);

// Check balances
console.log("Address1 Balance:", myToken.getBalance("Address1"));
console.log("Address2 Balance:", myToken.getBalance("Address2"));
console.log("Total Supply:", myToken.getTotalSupply());
